lib/ts/unit-tests/test_BufferWriterFormat.cc

```diff
@@ -232,6 +232,160 @@ TEST_CASE("bwstring", "[bwprint][bwstring]")
   REQUIRE(s == "32767 .. |e99a18c428cb38d5f260|");
 }
 
+TEST_CASE("BWFormat integral", "[bwprint][bwformat]")
+{
+  ts::LocalBufferWriter<256> bw;
+  ts::BWFSpec spec;
+  uint32_t num = 30;
+  int num_neg  = -30;
+
+  // basic
+  bwformat(bw, spec, num);
```

Comment:
Why call `bwformat` directly instead of via `BufferWriter::print`?

Reply:
I thought it was supposed to be testing the function specifically within the unit test. I added some tests with `print`. If you think it's better I can remove the `bwformat` ones and add more `print` tests.